src/js/components/Select/SelectContainer.js

```diff
@@ -382,11 +382,15 @@ const SelectContainer = forwardRef(
                   // as an option Button kind property.
                   let child;
                   if (children)
-                    child = children(option, index, options, {
-                      active: optionActive,
-                      disabled: optionDisabled,
-                      selected: optionSelected,
-                    });
+                    child = (
+                      <OptionBox selected={optionSelected}>
+                        {children(option, index, options, {
+                          active: optionActive,
+                          disabled: optionDisabled,
+                          selected: optionSelected,
+                        })}
+                      </OptionBox>
+                    );
                   else if (theme.select.options)
                     child = (
                       <OptionBox
```

Comment:
Wondering if we should reduce duplication and do `{child && <OptionBox ...>{child}</OptionBox>}` inside `<SelectOption />`?

Reply:
Since we are passing different props `selected={optionSelected}` in the first case and `selected={optionSelected}` and `{...selectOptionsStyle}` in the second case is it possible to reduce duplication?

Reply:
not in a significantly cleaner way